third-party/rsutils/include/rsutils/exceptions.h

```diff
@@ -0,0 +1,15 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+#pragma once
+
+
+namespace rsutils {
+
+
+// Enum that can be used to make code more verbose:
+//     ip_address( "blah", throw_if_not_valid );
+//
+enum _throw_if_not_valid { throw_if_not_valid };
```

Comment:
Maybe call the enum `may_throw`. We wont need the underline, and we could expand it to other throw reasons if needed

Reply:
I want it to say "an invalid value WILL throw", and not just "may throw" because it's not clear when.
The enum name and value name cannot be the same. I went for having the underscore on the type (definition) rather than the value (usage).
I can also add a `invalid_value_does_not_throw` and put it on the other ctor... up to you, but that would require changing other places, too.